Splitting option in file finder doesn't work if selected with Arrows + Enter

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Splitting option does nothing (not even open the file) if selected with arrow keys and enter.

- file finder: toggle
- use arrow up and down to select a file that isn't open yet
- press CMD to show split options
  - <img width="219" alt="image" src="https://github.com/user-attachments/assets/1ad97ccc-5615-4c69-9fd8-b8c9a7dbe593">
- use arrows to select option (see highlight)
  - <img width="201" alt="image" src="https://github.com/user-attachments/assets/778e1c84-75b1-4664-b8f8-d3524c82775a">
- press ENTER
- split options closes, nothing happens

It works if we click on the option or use the shown hotkeys.

If it is not supposed to be selectable in such way, I believe the options shouldn't be highlighted when we use the arrow keys.

Related to #20507, cc @SomeoneToIgnore

### Environment

Zed: v0.162.5 (Zed)
OS: macOS 14.7.1
Memory: 36 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2024-11-22T16:06:02.798636-03:00 [INFO] set status on client 145800: ConnectionLost
2024-11-22T16:06:02.799205-03:00 [INFO] set status on client 145800: Reauthenticating
2024-11-22T16:06:02.799412-03:00 [INFO] set status on client 145800: Reconnecting
2024-11-22T16:06:03.28489-03:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-11-22T16:06:04.573404-03:00 [INFO] add_connection;
2024-11-22T16:06:04.575129-03:00 [INFO] set status on client 145800: Connected { peer_id: PeerId { owner_id: 614, id: 1421401 }, connection_id: ConnectionId { owner_id: 0, id: 8 } }
2024-11-22T16:48:55.793455-03:00 [ERROR] connection error: delay between messages too long
2024-11-22T16:48:55.795526-03:00 [INFO] set status on client 145800: ConnectionLost
2024-11-22T16:48:55.796498-03:00 [INFO] set status on client 145800: Reauthenticating
2024-11-22T16:48:55.796938-03:00 [INFO] set status on client 145800: Reconnecting
2024-11-22T16:48:57.664467-03:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-11-22T16:48:58.812015-03:00 [INFO] add_connection;
2024-11-22T16:48:58.812908-03:00 [INFO] set status on client 145800: Connected { peer_id: PeerId { owner_id: 614, id: 1447216 }, connection_id: ConnectionId { owner_id: 0, id: 9 } }
2024-11-22T17:16:07.857285-03:00 [WARN] skipping diagnostics update, no worktree found for path "/Users/joaobezerra/.rustup/toolchains/1.81-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/time.rs"
2024-11-22T17:16:09.155606-03:00 [WARN] skipping diagnostics update, no worktree found for path "/Users/joaobezerra/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.7/src/builder/arg.rs"
2024-11-22T17:16:16.746217-03:00 [ERROR] error upgrading view
2024-11-22T17:19:56.066517-03:00 [WARN] skipping diagnostics update, no worktree found for path "/Users/joaobezerra/.rustup/toolchains/1.81-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/option.rs"
2024-11-22T17:19:56.694472-03:00 [WARN] skipping diagnostics update, no worktree found for path "/Users/joaobezerra/.rustup/toolchains/1.81-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/option.rs"
2024-11-22T17:20:01.158405-03:00 [WARN] skipping diagnostics update, no worktree found for path "/Users/joaobezerra/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.7/src/builder/arg.rs"
2024-11-22T17:21:14.87298-03:00 [ERROR] error upgrading view
2024-11-22T17:21:19.314016-03:00 [WARN] skipping diagnostics update, no worktree found for path "/Users/joaobezerra/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.7/src/builder/arg.rs"
2024-11-22T17:21:19.397048-03:00 [WARN] skipping diagnostics update, no worktree found for path "/Users/joaobezerra/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_derive-4.5.5/src/lib.rs"
2024-11-22T17:21:19.412758-03:00 [WARN] skipping diagnostics update, no worktree found for path "/Users/joaobezerra/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.7/src/builder/arg.rs"
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/assets/keymaps/default-linux.json
+++ b/assets/keymaps/default-linux.json
@@ -650,10 +650,15 @@
     }
   },
   {
+    "context": "FileFinder",
+    "bindings": {
+      "ctrl": "file_finder::ToggleMenu"
+    }
+  },
+  {
     "context": "FileFinder && !menu_open",
     "bindings": {
       "ctrl-shift-p": "file_finder::SelectPrev",
-      "ctrl": "file_finder::OpenMenu",
       "ctrl-j": "pane::SplitDown",
       "ctrl-k": "pane::SplitUp",
       "ctrl-h": "pane::SplitLeft",

--- a/assets/keymaps/default-macos.json
+++ b/assets/keymaps/default-macos.json
@@ -651,10 +651,15 @@
     }
   },
   {
+    "context": "FileFinder",
+    "bindings": {
+      "cmd": "file_finder::ToggleMenu"
+    }
+  },
+  {
     "context": "FileFinder && !menu_open",
     "bindings": {
       "cmd-shift-p": "file_finder::SelectPrev",
-      "cmd": "file_finder::OpenMenu",
       "cmd-j": "pane::SplitDown",
       "cmd-k": "pane::SplitUp",
       "cmd-h": "pane::SplitLeft",

--- a/crates/file_finder/src/file_finder.rs
+++ b/crates/file_finder/src/file_finder.rs
@@ -42,7 +42,7 @@ use workspace::{
     Workspace,
 };
 
-actions!(file_finder, [SelectPrev, OpenMenu]);
+actions!(file_finder, [SelectPrev, ToggleMenu]);
 
 impl ModalView for FileFinder {
     fn on_before_dismiss(&mut self, cx: &mut ViewContext<Self>) -> workspace::DismissDecision {
@@ -55,7 +55,6 @@ impl ModalView for FileFinder {
 
 pub struct FileFinder {
     picker: View<Picker<FileFinderDelegate>>,
-    picker_focus_handle: FocusHandle,
     init_modifiers: Option<Modifiers>,
 }
 
@@ -156,14 +155,8 @@ impl FileFinder {
     }
 
     fn new(delegate: FileFinderDelegate, cx: &mut ViewContext<Self>) -> Self {
-        let picker = cx.new_view(|cx| Picker::uniform_list(delegate, cx));
-        let picker_focus_handle = picker.focus_handle(cx);
-        picker.update(cx, |picker, _| {
-            picker.delegate.focus_handle = picker_focus_handle.clone();
-        });
         Self {
-            picker,
-            picker_focus_handle,
+            picker: cx.new_view(|cx| Picker::uniform_list(delegate, cx)),
             init_modifiers: cx.modifiers().modified().then_some(cx.modifiers()),
         }
     }
@@ -189,10 +182,12 @@ impl FileFinder {
         cx.dispatch_action(Box::new(menu::SelectPrev));
     }
 
-    fn handle_open_menu(&mut self, _: &OpenMenu, cx: &mut ViewContext<Self>) {
+    fn handle_toggle_menu(&mut self, _: &ToggleMenu, cx: &mut ViewContext<Self>) {
         self.picker.update(cx, |picker, cx| {
             let menu_handle = &picker.delegate.popover_menu_handle;
-            if !menu_handle.is_deployed() {
+            if menu_handle.is_deployed() {
+                menu_handle.hide(cx);
+            } else {
                 menu_handle.show(cx);
             }
         });
@@ -265,8 +260,8 @@ impl FileFinder {
 impl EventEmitter<DismissEvent> for FileFinder {}
 
 impl FocusableView for FileFinder {
-    fn focus_handle(&self, _: &AppContext) -> FocusHandle {
-        self.picker_focus_handle.clone()
+    fn focus_handle(&self, cx: &AppContext) -> FocusHandle {
+        self.picker.focus_handle(cx)
     }
 }
 
@@ -282,7 +277,7 @@ impl Render for FileFinder {
             .w(modal_max_width)
             .on_modifiers_changed(cx.listener(Self::handle_modifiers_changed))
             .on_action(cx.listener(Self::handle_select_prev))
-            .on_action(cx.listener(Self::handle_open_menu))
+            .on_action(cx.listener(Self::handle_toggle_menu))
             .on_action(cx.listener(Self::go_to_file_split_left))
             .on_action(cx.listener(Self::go_to_file_split_right))
             .on_action(cx.listener(Self::go_to_file_split_up))
@@ -308,7 +303,6 @@ pub struct FileFinderDelegate {
     separate_history: bool,
     first_update: bool,
     popover_menu_handle: PopoverMenuHandle<ContextMenu>,
-    focus_handle: FocusHandle,
 }
 
 /// Use a custom ordering for file finder: the regular one
@@ -638,7 +632,6 @@ impl FileFinderDelegate {
             separate_history,
             first_update: true,
             popover_menu_handle: PopoverMenuHandle::default(),
-            focus_handle: cx.focus_handle(),
         }
     }
 
@@ -1263,11 +1256,7 @@ impl PickerDelegate for FileFinderDelegate {
                         .trigger(
                             Button::new("actions-trigger", "Split Options")
                                 .selected_label_color(Color::Accent)
-                                .key_binding(KeyBinding::for_action_in(
-                                    &OpenMenu,
-                                    &self.focus_handle,
-                                    cx,
-                                )),
+                                .key_binding(KeyBinding::for_action(&ToggleMenu, cx)),
                         )
                         .menu({
                             move |cx| {

--- a/crates/file_finder/src/file_finder.rs
+++ b/crates/file_finder/src/file_finder.rs
@@ -55,6 +55,7 @@ impl ModalView for FileFinder {
 
 pub struct FileFinder {
     picker: View<Picker<FileFinderDelegate>>,
+    picker_focus_handle: FocusHandle,
     init_modifiers: Option<Modifiers>,
 }
 
@@ -155,8 +156,14 @@ impl FileFinder {
     }
 
     fn new(delegate: FileFinderDelegate, cx: &mut ViewContext<Self>) -> Self {
+        let picker = cx.new_view(|cx| Picker::uniform_list(delegate, cx));
+        let picker_focus_handle = picker.focus_handle(cx);
+        picker.update(cx, |picker, _| {
+            picker.delegate.focus_handle = picker_focus_handle.clone();
+        });
         Self {
-            picker: cx.new_view(|cx| Picker::uniform_list(delegate, cx)),
+            picker,
+            picker_focus_handle,
             init_modifiers: cx.modifiers().modified().then_some(cx.modifiers()),
         }
     }
@@ -260,8 +267,8 @@ impl FileFinder {
 impl EventEmitter<DismissEvent> for FileFinder {}
 
 impl FocusableView for FileFinder {
-    fn focus_handle(&self, cx: &AppContext) -> FocusHandle {
-        self.picker.focus_handle(cx)
+    fn focus_handle(&self, _: &AppContext) -> FocusHandle {
+        self.picker_focus_handle.clone()
     }
 }
 
@@ -303,6 +310,7 @@ pub struct FileFinderDelegate {
     separate_history: bool,
     first_update: bool,
     popover_menu_handle: PopoverMenuHandle<ContextMenu>,
+    focus_handle: FocusHandle,
 }
 
 /// Use a custom ordering for file finder: the regular one
@@ -632,6 +640,7 @@ impl FileFinderDelegate {
             separate_history,
             first_update: true,
             popover_menu_handle: PopoverMenuHandle::default(),
+            focus_handle: cx.focus_handle(),
         }
     }
 
@@ -1235,6 +1244,7 @@ impl PickerDelegate for FileFinderDelegate {
     }
 
     fn render_footer(&self, cx: &mut ViewContext<Picker<Self>>) -> Option<AnyElement> {
+        let context = self.focus_handle.clone();
         Some(
             h_flex()
                 .w_full()
@@ -1256,15 +1266,19 @@ impl PickerDelegate for FileFinderDelegate {
                         .trigger(
                             Button::new("actions-trigger", "Split Options")
                                 .selected_label_color(Color::Accent)
-                                .key_binding(KeyBinding::for_action(&ToggleMenu, cx)),
+                                .key_binding(KeyBinding::for_action_in(&ToggleMenu, &context, cx)),
                         )
                         .menu({
                             move |cx| {
-                                Some(ContextMenu::build(cx, move |menu, _| {
-                                    menu.action("Split Left", pane::SplitLeft.boxed_clone())
-                                        .action("Split Right", pane::SplitRight.boxed_clone())
-                                        .action("Split Up", pane::SplitUp.boxed_clone())
-                                        .action("Split Down", pane::SplitDown.boxed_clone())
+                                Some(ContextMenu::build(cx, {
+                                    let context = context.clone();
+                                    move |menu, _| {
+                                        menu.context(context)
+                                            .action("Split Left", pane::SplitLeft.boxed_clone())
+                                            .action("Split Right", pane::SplitRight.boxed_clone())
+                                            .action("Split Up", pane::SplitUp.boxed_clone())
+                                            .action("Split Down", pane::SplitDown.boxed_clone())
+                                    }
                                 }))
                             }
                         }),

--- a/crates/ui/src/components/keybinding.rs
+++ b/crates/ui/src/components/keybinding.rs
@@ -3,7 +3,7 @@ use crate::PlatformStyle;
 use crate::{h_flex, prelude::*, Icon, IconName, IconSize};
 use gpui::{relative, Action, FocusHandle, IntoElement, Keystroke, WindowContext};
 
-#[derive(IntoElement, Clone)]
+#[derive(Debug, IntoElement, Clone)]
 pub struct KeyBinding {
     /// A keybinding consists of a key and a set of modifier keys.
     /// More then one keybinding produces a chord.

